### ISSUE TITLE
Ubuntu 16.04 build, Unicode output on fizmo-ncursesw vs. fizmo-sdl2

### ISSUE THREAD

**Issue by BroadcastGames** [chrender/fizmo#5]
Hello. Ubuntu 16.04 x64 build of latest git source code checkout 196afec6d8946e8b668826da78402e6fd8e29af5

I made a very simple Inform 7 sample to compile to .z8 for testing Unicode output. On Ubuntu 16.04, the Inform 7 IDE runs this fine with it's built-in interpreter and correctly displays the Unicode output. https://github.com/BroadcastGames/Inform7_StoryCode0/blob/master/storyFilesPile0/UnicodeZMachineA.ni

fizmo-nucrsesw built on the same system correctly shows all the Unicode characters. fizmo-sdl2 shows boxes instead of the characters.

For reference:the same issue appears with Gargoyle Glk https://github.com/garglk/garglk not showing the Unicode built from source code on the same system.  garglk provides an interesting example because you can copy/paste the text out of Gargoyle into Gedit and it shows all the Unicode correctly.  So it's some kind of font issue?  Since the Inform 7 IDE for GNOME can correctly show the output on the same system - perhaps some hint of how they solved the font issue lies in that source code? Thank you.

**Comment by chrender** [comment]
I've been looking into this and it indeed appears to be a font issue. Fizmo currently comes pre-packaged with the [Fira font family](https://www.mozilla.org/en-US/styleguide/products/firefox-os/typeface/), which doesn't come with any arabic or other non-western glyphs.

I've been experimenting with the [Google Noto font family](https://www.google.com/get/noto/) which also provides a font with arabic glyphs. Using this one, your test case correctly displays the missing arabic glyphs. So in order to solve the display problem, I'll have to support loading multiple font files for a single font face.

I've been skimming over the Z-Spec to determine which kind of regions / languages should be supported, but so far I haven't found a specific list. Are you looking to support specific languages or is your test case looking at specific glyphs because you need these?

**Comment by BroadcastGames** [comment]
In my situation, it's probably fine as is - because fizmo is only Z-machine code, right? But I was hoping to use some of it's SDL2 code to get nanoglk updated from SDL 1.2 to 2.0 and solve some of these same problems. In terms of encountering mixed Unicode fonts in the wild - the Inform 7 IDE on Ubuntu has no trouble showing all these mixed languages with either Z-machine or Gluxe stories.  Gargoyle (GarGlk) has font display problems on

In terms of fixing it, I'm entirely fine with it being low priority (unsolved for now). Again, because Z-machine only I"m not likely to encounter it with newly authored Inform stories which will more likely target Glulx. Leave the issue open for the future if someone wants to solve it and so people know the issue is there?

**Comment by chrender** [comment]
Agreed, at the moment it will be better to resolve the remaining issues in order to release version 0.8.4. For the next version I'll see whether it is possible to support multiple font files simultaneously, which would allow to solve this issue using existing fonts.

This issue will be “on hold” for the time being.
